src/scripts/main.js

```diff
@@ -1,3 +1,262 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('.button');
+const startMessage = document.querySelector('.message-start');
+const loseMessage = document.querySelector('.message-lose');
+const allMessages = document.querySelectorAll('.message');
+const winMessage = document.querySelector('.message-win');
+const scoreCountElement = document.querySelector('.game-score');
+
+const fieldRows = document.querySelectorAll('.field-row');
+
+const firstRow = fieldRows[0].children;
+const secondRow = fieldRows[1].children;
+const thirdRow = fieldRows[2].children;
+const fourthRow = fieldRows[3].children;
+
+let scoreOfTheGame = 500;
+let theGameStarted = false;
+
+const gameMarkup = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+startButton.addEventListener('click', () => {
+  if (startButton.classList.contains('start')) {
+    theGameStarted = true;
+
+    startButton.classList.remove('start');
+    startButton.classList.add('restart');
+    startButton.textContent = 'Restart';
+
+    scoreOfTheGame = 0;
+    startMessage.classList.add('hidden');
+
+    gameMarkup.map(row => row.fill(0));
+
+    fillRandomCell();
+    fillRandomCell();
+    render();
+  } else if (startButton.classList.contains('restart')) {
+    if (!checkGameOver()) {
+      scoreOfTheGame = 0;
+
+      gameMarkup.map(row => row.fill(0));
+
+      fillRandomCell();
+      fillRandomCell();
+      render();
+    } else {
+      startButton.classList.remove('restart');
```

Comment:
You can just break the function to avoid nesting
```suggestion
      fillRandomCell();
      render();
      return;
    }
    
    startButton.classList.remove('restart');
```